pyctcdecode/alphabet.py

```diff
@@ -3,37 +3,74 @@
 from typing import List, Optional
 
 
+UNK_CHAR = "⁇"  # representation of unknown character in regular alphabet
 BPE_CHAR = "▁"  # character used for token boundary if BPE is used
 UNK_BPE_CHAR = "▁⁇▁"  # representation of unknown character in BPE
 
 logger = logging.getLogger(__name__)
 
 
-def _get_ctc_index(label_list: List[str]) -> int:
-    """Get index of ctc blank character in alphabet."""
-    return len(label_list) - 1 if label_list[-1] == "" else -1
-
-
-def _normalize_alphabet(label_list: List[str], ctc_token_idx: Optional[int] = None) -> List[str]:
-    """Normalize alphabet for non-bpe decoder."""
-    if any([len(c) > 1 for c in label_list]):
-        raise ValueError("For non-bpe alphabet only length 1 entries and blank token are allowed.")
-    if ctc_token_idx is None:
-        ctc_token_idx = _get_ctc_index(label_list)
-    clean_labels = label_list[:]
-    # check for space token
-    if " " not in clean_labels:
-        raise ValueError("Space token ' ' missing from vocabulary.")
-    # specify ctc blank token
-    if ctc_token_idx == -1:
-        clean_labels.append("")
+def _check_if_bpe(labels: List[str]):
+    """Check if input alphabet is BPE or not."""
+    if any([s.startswith("##") for s in labels]) or any([s.startswith(BPE_CHAR) for s in labels]):
+        logger.info("Alphabet determined to be of BPE style.")
+        is_bpe = True
     else:
-        clean_labels[ctc_token_idx] = ""
-    return clean_labels
+        logger.info("Alphabet determined to be of regular style.")
+        is_bpe = False
+    return is_bpe
+
+
+def _normalize_regular_alphabet(labels: List[str]) -> List[str]:
+    """Normalize non-bpe labels to alphabet for decoder."""
+    normalized_labels = labels[:]
+    # substitute space characters
+    if "|" in normalized_labels and " " not in normalized_labels:
+        logger.info("Found '|' in vocabulary but not ' ', doing substitution.")
+        normalized_labels[normalized_labels.index("|")] = " "
+    elif "|" in normalized_labels:
+        logger.warning(
+            "Found '|' in vocabulary. If this denotes something special, please construct the "
+            "alphabet manually."
+        )
+    # substituted ctc blank char
+    if "_" in normalized_labels and "" not in normalized_labels:
+        logger.info("Found '_' in vocabulary but not '', doing substitution.")
```

Comment:
at the very least, all of the logging statements in this method should probably be warnings, because we're trying to patch up malformed input and that's always ripe for errors

Reply:
so my thoughts were that this is basically a convenience method for assembling the alphabet. meaning substitutions are actually expected rather than outliers. (for example appending ctc blank char, which conventionally isn't included in the labels). If you want to do it manually you can just instantiate the object directly.
Thoughts?

Reply:
ok, I guess that makes sense.